hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -328,6 +341,43 @@ public ResponseCodeEnum adjustBalance(AccountID aId, TokenID tId, long adjustmen
 		return sanityChecked(aId, tId, token -> tryAdjustment(aId, tId, adjustment));
 	}
 
+	@Override
+	public ResponseCodeEnum adjustBalance(AccountID senderAId, AccountID receiverAId, TokenID tId, long serialNumber) {
+		var validity = sanityChecked(senderAId, receiverAId, tId, token -> tryAdjustment(senderAId, receiverAId, tId, serialNumber));
+		if (validity == OK) {
+			var nftId = NftID.newBuilder().setTokenID(tId).setSerialNumber(serialNumber).build();
+			if (nftExists(nftId)) {
+				get(nftId).setOwner(EntityId.fromGrpcAccountId(receiverAId));
+
+				adjustOwnedNfts(senderAId, false);
```

Comment:
This seems quite risky, as `adjustOwnedNfts()` is irreversibly changing the mutable leaf right here, while the surrounding code is just adding "provisional" changes in the `TransationalLedger` proxy---which in principle could be rolled back.

Probably a non-issue given the ongoing refactor, however.